dataServices.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const products = require('./src/data/products');
+const categories = require('./src/data/categories');
+
+let productMap = null;
+let categoryMap = null;
+let combinedProductMap = null;
+
+module.exports = {
+  getProducts: () => {
+    if (productMap === null) {
+      productMap = {};
+      products.products.forEach(product => {
+        productMap[product.id] = product;
+      });
+    }
+    return productMap;
+  },
+  getCategories: () => {
+    if (categoryMap === null) {
+      categoryMap = {};
+      categories.categories.forEach(category => {
+        categoryMap[category.id] = category;
+      });
+    }
+    return categoryMap;
+  },
+  getCombinedProductMap: () => {
+    if (combinedProductMap === null) {
+      module.exports.combineProductsWithCategories();
```

Comment:
What is the use of this `module.exports..`? 

Migrate all the function out of export, and call it as a simple functions